src/ethereum/berlin/vm/precompiled_contracts/modexp.py

```diff
@@ -31,25 +31,23 @@ def modexp(evm: Evm) -> None:
     exp_length = U256.from_be_bytes(right_pad_zero_bytes(data[32:64], 32))
     modulus_length = U256.from_be_bytes(right_pad_zero_bytes(data[64:96], 32))
 
+    exponent_head_start = 96 + base_length
+    exponent_head_end = exponent_head_start + min(32, exp_length)
+
+    exponent_head_bytes = data[exponent_head_start:exponent_head_end]
+    exponent_head = Uint.from_be_bytes(exponent_head_bytes)
+
+    gas_used = gas_cost(base_length, modulus_length, exp_length, exponent_head)
+
+    if gas_used > U256.MAX_VALUE:
```

Comment:
Isn't this more of a `ValueError` than a `OutOfGasError` at this point?

Reply:
Good question. I think the tests expect execution to continue as if it had run out of gas, but haven't specifically checked.

Reply:
An alternative would be to do what [Geth does](https://github.com/ethereum/go-ethereum/blob/714fb302a5f1c19dc2fc3973d01210f4e2624a91/core/vm/contracts.go#L357) and cap the `gas_used` to `Uint64.MAX_VALUE` and let the `subtract_gas` function raise the `OutOfGasError` if applicable.